src/graphics/graphics-device.js

```diff
@@ -224,12 +224,18 @@ class GraphicsDevice extends EventHandler {
     canvas;
 
     /**
-     * The highest shader precision supported by this graphics device. Can be 'hiphp', 'mediump' or
-     * 'lowp'.
+     * The WebGL context managed by this graphics device.
      *
-     * @type {string}
+     * @type {WebGLRenderingContext}
```

Comment:
Do you intent to not include `WebGL2RenderingContext` constants? Whatever is decided upon, it would be a bit tricky to remember what was decided upon. Maybe instead of checking `if (this.webgl2)` there could be another context variable called `this.gl1` and then it can have the correct type and the if-check just becomes `if (this.gl1)` e.g.

Reply:
Thanks!

Reply:
So I had a play with this last night by setting the type to `WebGLRenderingContext|WebGL2RenderingContext ` and that didn't work. What you suggest (maintaining two context properties) would be a fairly significant change so, for now, I thought it would be useful to just enable 90% of the WebGL calls to work with IntelliSense. Supporting WebGL2 can be handled in a follow up PR if we decide on a good path forwards there.

Reply:
fair decision - perhaps capture some of it as a comment in the code please.

Reply:
Done. And I've just switched the type to `WebGL2RenderingContext` which allows all WebGL 2.0 calls to work with IntelliSense too. See comment for justification.

Reply:
> So I had a play with this last night by setting the type to `WebGLRenderingContext|WebGL2RenderingContext ` and that didn't work.

You need an intersection via `WebGLRenderingContext & WebGL2RenderingContext`

https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types

Reply:
> WebGL 2 is a superset of WebGL1 so `WebGL2RenderingContext` should be the same as `WebGLRenderingContext & WebGL2RenderingContext` right?

Yea, thats what I also think, they stick to "extending" by only adding more optional arguments to all kinds of functions, e.g. for `readPixels`:

https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L16131-L16133

https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L16983

At least I didn't see one function so far in `WebGLRenderingContext` that doesn't exist in some form (with optional arguments often) in `WebGL2RenderingContext` too, so as it is right now, its all good